protocol-definitions/custom/Custom.yaml

```diff
@@ -868,5 +868,44 @@ customTypes:
           type: Data
           nullable: true
           since: 2.0
-
-
+  - name: SqlError
+    since: 2.1
+    params:
+      - name: code
+        type: int
+        nullable: false
+        since: 2.1
+      - name: message
+        type: String
+        nullable: true
+        since: 2.1
+      - name: originatingMemberId
+        type: UUID
+        nullable: false
+        since: 2.1
+  - name: SqlColumnMetadata
+    since: 2.1
+    returnWithFactory: true
+    params:
+      - name: name
+        type: String
+        nullable: false
+        since: 2.1
+      - name: type
+        type: int
+        nullable: false
+        since: 2.1
+  - name: SqlRowMetadata
+    since: 2.1
+    params:
+      - name: columns
+        type: List_SqlColumnMetadata
+        nullable: false
+        since: 2.1
+  - name: SqlRow
```

Comment:
if this is a single parameter custom type, you can unwrap it

Reply:
I tried to remove the `SqlRow` from the protocol. If I do so, I would have to introduce the new `List_List_Data` type, because currently there is `List_SqlRow` as well. 

The problem is that in the future we will not pass row content as `List_Data`. Instead, it will be some form of `SqlRow` with some different content (e.g. specially encoded `SqlRowValue` objects or so). When we do this change, `List_List_Data` will be unused. So IMO it is better to use `SqlRow` for now, because it will not lead to unused types in the future.

WDYT?

Reply:
First of all, client protocol doesn't support changing the custom type. We can only add more parameters.
If you are planning to update a Data type with a full custom encoding then, you need to update method id.
List_List_Data is currently supported, all you need to do is add it to init list like others. The code generator will handle the generation and no new codec will be added.
it will generate:
        ListMultiFrameCodec.encode(clientMessage, listDataData, ListDataCodec::encode);
where ListMultiFrameCodec and ListDataCodec  are already in the codebase.
When you need to update List_List_Data to List_SqlRow you just delete List_List_Data if it's not used elsewhere.
please note that List_List_Data is some generic codec that CP generator handles smartly.
From pure CP best practices, it would be better to add an actual SqlRow with actual fields later on.